v2/kafka-to-bigquery/src/test/java/com/google/cloud/teleport/v2/templates/KafkaToBigQueryFlexIT.java

```diff
@@ -0,0 +1,234 @@
+    Schema dlqSchema =
+        Schema.of(
+            Field.newBuilder("timestamp", StandardSQLTypeName.TIMESTAMP)
+                .setMode(Mode.REQUIRED)
+                .build(),
+            Field.newBuilder("payloadString", StandardSQLTypeName.STRING)
+                .setMode(Mode.REQUIRED)
+                .build(),
+            Field.newBuilder("payloadBytes", StandardSQLTypeName.BYTES)
+                .setMode(Mode.REQUIRED)
+                .build(),
+            Field.newBuilder(
+                    "attributes",
+                    StandardSQLTypeName.STRUCT,
+                    Field.newBuilder("key", StandardSQLTypeName.STRING)
+                        .setMode(Mode.NULLABLE)
+                        .build(),
+                    Field.newBuilder("value", StandardSQLTypeName.STRING)
+                        .setMode(Mode.NULLABLE)
+                        .build())
+                .setMode(Mode.REPEATED)
+                .build(),
+            Field.newBuilder("errorMessage", StandardSQLTypeName.STRING)
+                .setMode(Mode.NULLABLE)
+                .build(),
+            Field.newBuilder("stacktrace", StandardSQLTypeName.STRING)
+                .setMode(Mode.NULLABLE)
+                .build());
+    return dlqSchema;
+  }
+
+  public void baseKafkaToBigQuery(Function<LaunchConfig.Builder, LaunchConfig.Builder> paramsAdder)
+      throws IOException {
+    // Arrange
+    String topicName = kafkaResourceManager.createTopic(testName, 5);
+
+    String bqTable = testName;
+    Schema bqSchema =
+        Schema.of(
+            Field.of("id", StandardSQLTypeName.INT64),
+            Field.of("name", StandardSQLTypeName.STRING));
+
+    TableId tableId = bigQueryClient.createTable(bqTable, bqSchema);
+    TableId deadletterTableId = TableId.of(tableId.getDataset(), tableId.getTable() + "_dlq");
+
+    LaunchConfig.Builder options =
+        paramsAdder.apply(
+            LaunchConfig.builder(testName, specPath)
+                .addParameter(
+                    "readBootstrapServerAndTopic",
+                    kafkaResourceManager.getBootstrapServers().replace("PLAINTEXT://", "")
+                        + ";"
+                        + topicName)
+                .addParameter("kafkaReadOffset", "earliest")
+                .addParameter("outputTableSpec", toTableSpecLegacy(tableId))
+                .addParameter("outputDeadletterTable", toTableSpecLegacy(deadletterTableId)));
+
+    // Act
+    LaunchInfo info = launchTemplate(options);
+    assertThatPipeline(info).isRunning();
+    KafkaProducer<String, String> kafkaProducer =
+        kafkaResourceManager.buildProducer(new StringSerializer(), new StringSerializer());
+
+    for (int i = 1; i <= 10; i++) {
+      publish(kafkaProducer, topicName, i + "1", "{\"id\": " + i + "1, \"name\": \"Dataflow\"}");
+      publish(kafkaProducer, topicName, i + "2", "{\"id\": " + i + "2, \"name\": \"Pub/Sub\"}");
+      // Invalid schema
```

Comment:
why is this invalid?

Reply:
The schema is different. These records are there to tests the DLQ.

Reply:
Okay. can we make the comment clear? Thanks

Reply:
Done